File: 462-minimum-moves-to-equal-array-elements-ii/462-minimum-moves-to-equal-array-elements-ii.cpp
class Solution {
public:
    int minMoves2(vector<int>& nums) {
        return daily(nums);
    }
private:
    // O(n logn) Time and O(1) extra space
    int daily(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        int n = nums.size();
        int req = nums[n / 2]; 
        
        int ans = 0;
        
        for(int i = 0; i < n; ++i) {
            ans += abs(nums[i] - req);
        }
        
        return ans;
    }
};

File: 462-minimum-moves-to-equal-array-elements-ii/462-minimum-moves-to-equal-array-elements-ii.py
class Solution:
    def minMoves2(self, nums: List[int]) -> int:
        n = len(nums)
        
        nums.sort()
        req = nums[n // 2]
                
        ans = 0
        
        for x in nums:
            ans += abs(x - req)
        return ans